File: tasks/vulcancyber-v1/oss-content-disposition-ext-value/tests/vb_regression.test.ts
// Hidden pass_to_pass regression guard for oss-content-disposition-ext-value.
//
// Rejecting malformed escapes must not change parsing of well-formed values: a
// valid UTF-8 ext-value still decodes, and a plain quoted filename is unchanged.
// Both hold at the base commit and after the fix. Run with `tsx --test`.

import { test } from 'node:test'
import assert from 'node:assert'
import { parse } from './src/index.ts'

test('vb valid utf-8 ext-value still decodes', () => {
  const p = parse("attachment; filename*=utf-8''%E2%82%AC%20rates.pdf").parameters
  assert.strictEqual(p.filename, '€ rates.pdf')
})

test('vb plain quoted filename is unchanged', () => {
  const p = parse('attachment; filename="a.pdf"').parameters
  assert.strictEqual(p.filename, 'a.pdf')
})

File: tasks/vulcancyber-v1/oss-echo-encoded-path-separator/tests/vbcheck/vb_regression_test.go
// SPDX-License-Identifier: MIT
// Hidden pass_to_pass regression guard for oss-echo-encoded-path-separator.
// Rejecting encoded separators must not break ordinary static serving: a normal
// static file is still served, and a plain (unencoded) path into the protected
// group is still handled by the group's middleware. Both hold at the base commit
// and after the fix. External package: uses only echo's public API.

package vbcheck

import (
	"net/http"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestVbLegitStaticStillServed(t *testing.T) {
	code, body := serve(newProtectedApp(), "/index.html")
	assert.Equal(t, http.StatusOK, code, "a legitimate static file must still be served")
	assert.Equal(t, "public", body)
}

// An encoded backslash decodes to a literal filename character (not a separator on
// fs.FS), so it resolves nothing and is already Not Found — guard that this stays
// true and never regresses into serving a file.
func TestVbEncodedBackslashStaysNotFound(t *testing.T) {
	code, body := serve(newProtectedApp(), "/admin%5Csecret.txt")
	assert.Equal(t, http.StatusNotFound, code)
	assert.NotContains(t, body, "TOP-SECRET")
}
